### PR TITLE
VAGOV-6366 add link reporting

### PR DESCRIPTION
**Pull request by swirtSJW**
Testing Steps:
Vist existing content with diverse links validate that node report shows  and check for appropriateness of the report:
- [x] (lots of links) /pittsburgh-health-care/locations/belmont-county-va-clinic
- [x] (Lots of links, odd lnks to blocks) /health-care
- [x] (link to non drupal page) /health-care/about-va-health-benefits/vision-care/blind-low-vision-rehab-services
- [x] (911 link fatal error) /pittsburgh-health-care/operating-status

Create a node of any content type that has fields that support links.

- [x] insert link to published internal node
- [x] insert link to unpublished node
- [x] insert ink to valid external site (https://amazon.com)
- [x] insert link to non-existent site (https://weatherunderBound.com)
- [x]  save the node.  Validate that node report shows up showing one broken link and one unpublished link.
- [x] Edit the node and validate that the node report shows on node edit.
- [x] add a newlink to a broken external link (https://no-site-to=-e-found-here.gov)
- [x] save the node and validate that now there are two broken links. and 1 unpublished warning link

### CODE REVIEW

**Comment by swirtSJW** [comment]
The behat catch is a valid one.   It has identified that there are a couple of pages that completely blow up when trying to display the link report.

I haven't found the connection yet but these are the pages I am investigating that fatal error:

- /node/36   /health-care/order-hearing-aid-batteries-prosthetic-socks
- /node/936  /education/about-gi-bill-benefits/post-9-11/yellow-ribbon-program
- /node/586  /pension/how-to-apply
- /node/2 /health-care/about-va-health-benefits/vision-care/blind-low-vision-rehab-services

> The website encountered an unexpected error. Please try again later.</br></br><em class="placeholder">UnexpectedValueException</em>: External URLs do not have internal route parameters. in <em class="placeholder">Drupal\Core\Url-&gt;getRouteParameters()</em> (line <em class="placeholder">571</em> of <em class="placeholder">core/lib/Drupal/Core/Url.php</em>).

**Comment by swirtSJW** [comment]
I figured it out... it is when it is trying to look up internal links that are not managed by drupal, and unavailable to the cms.  I am putting in a pre-check to keep it from failing catastrophically.

**Comment by swirtSJW** [comment]
I just updated the contrib to handle the fatal errors.

**Comment by swirtSJW** [comment]
Noticed an issue.... converting to WIP

**Comment by swirtSJW** [comment]
Double checked... it was a non-issue.  Removed wip

**Comment by kevwalsh** [comment]
@swirtSJW 
Before starting from scratch with new content, i logged in as randi.hecht@va.gov and clicked the most recent 4 nodes on /admin/content

**False positive:**
http://pr811.ci.cms.va.gov/careers-employment/vocational-rehabilitation/how-to-apply 
http://pr811.ci.cms.va.gov/health-care/about-va-health-benefits/vision-care/blind-low-vision-rehab-services

**504 gateway timeout (possibly unrelated)**
http://pr811.ci.cms.va.gov/health-care 

**Issue with `tel:` link checker?**
http://pr811.ci.cms.va.gov/pittsburgh-health-care/operating-status


Is this PR enviro pr811 missing your most recent changes?

**Comment by swirtSJW** [comment]
It looks like that environment is not using the latest version of the module... likely picking it up from composer cache....   Investigating.

**Comment by swirtSJW** [comment]
Actually, it is using the right version..... digging deeper.

**Comment by swirtSJW** [comment]
Converting this back to wip while I work out some scalability issues with CI and BRD trying to do serial curls.   Need to change it to multi_culr for parallel processing.

**Comment by swirtSJW** [comment]
Leaving this as a WIP while I resolve catastphic failures on bad urls

File: composer.lock [hunk 1]
```diff
@@ -5806,8 +5806,8 @@
                     "dev-3.x": "3.x-dev"
                 },
                 "drupal": {
-                    "version": "8.x-3.0-rc1+15-dev",
-                    "datestamp": "1570894385",
+                    "version": "8.x-3.0-beta1+54-dev",
+                    "datestamp": "1553812381",
```
Comment (swirtSJW) [inline composer.lock]:
```suggestion
                    "datestamp": "1570894385",
```

File: composer.lock [hunk 2]
```diff
@@ -5806,8 +5806,8 @@
                     "dev-3.x": "3.x-dev"
                 },
                 "drupal": {
-                    "version": "8.x-3.0-rc1+15-dev",
-                    "datestamp": "1570894385",
+                    "version": "8.x-3.0-beta1+54-dev",
```
Comment (swirtSJW) [inline composer.lock]:
```suggestion
                    "version": "8.x-3.0-rc1+15-dev",
```

**Pull request merged**